app/scripts/controllers/reviewRegistration.js

```diff
@@ -249,8 +249,13 @@ angular.module('confRegistrationWebApp')
     $scope.mergeAndConfirmRegistration = function () {
       // Pay for the spouse's registration before merging it with the spouse
       $scope.submittingRegistration = true;
-      registration.submitPayment($scope.currentPayment, $scope.spouseRegistration, $scope.acceptedPaymentMethods()).then(function () {
-        return registration.mergeWithSpouse(currentRegistration, $scope.spouseRegistration);
+
```

Comment:
This commit successfully merges and then submits payment. Note that I pass `mergedRegistration` to `registration.submitPayment`.

I realize that the merge will take place even if payment fails (since we have no 'undo merge'). But I don't know of any other solution since full payments prior to merge will also fail.

Please review and comment @canac and/or @OzzieOrca